website/docs/targeting/feature-flag-evaluation.md

```diff
@@ -0,0 +1,125 @@
+---
+id: feature-flag-evaluation
+title: Feature Flag Evaluation
+description: This document offers an in-depth explanation of how the ConfigCat SDK determines the value of a feature flag.
+---
+
+# Feature Flag Evaluation
+
+This document offers an in-depth explanation of how the SDK determines the value of a feature flag when executing the `GetValue` function. Understanding this process requires prior knowledge of targeting concepts.
+
+The feature flag's value is influenced by:
+
+- The feature flag's rules defined on the Dashboard,
+- The [User Object] provided to the `GetValue` function, and
+- The *default value* passed to the `GetValue` function.
+
+The feature flag's value always comes from exactly one rule, following this algorithm:
+
+1. **Evaluation of Targeting Rules:** If targeting rules are present, the SDK evaluates them one by one, from top to bottom. It checks if all the conditions in the rule's IF part are met (i.e. all the conditions evaluate to true).
+   - If the conditions are met, the THEN part determines the value to return. Note: If the THEN part contains [Percentage Options] and the necessary user attribute is missing, the SDK will skip the targeting rule and continue with the next rule - even though the targeting rule's conditions are met!
+   - If the conditions aren't met, the SDK moves to the next targeting rule, or to step 2 (below) if there are no more targeting rules.
+2. **Evaluation of Percentage Options:** If a percentage option rule exists, the SDK executes the [Evaluation of percentage options] algorithm to determine which percentage option applies to the user and returns the value associated with that option. If the necessary user attribute is missing, the SDK skips to step 3 (below).
+3. **Returning simple value:** At this stage, the only remaining "rule" is the simple value specified at the end of the feature flag, which the SDK then returns.
+
+In the event of an unexpected error during evaluation, the SDK returns the default value passed to the `GetValue` function.
+
+## Evaluation of a Targeting Rule
+
+The SDK evaluates the conditions in the rule's IF part one by one, from top to bottom. The result of a condition can be one of the following: `true`, `false` or `cannot evaluate`.
+
+The result `cannot evaluate` occurs when the necessary user attribute is missing, invalid or incorrectly formatted (the SDK logs these issues as warnings).
+
+A targeting rule matches only when all its conditions evaluates to `true`. In any other cases, it doesn't match.
+
+### Evaluation of a User Condition
+
+The SDK looks up the comparison attribute (the user attribute referenced by the condition) in the [User Object]. It compares it to the comparison value that is set on the Dashboard. The comparison is done according to the selected comparator, resulting in a `true` or `false` value. This will be the result of the condition.
```

Comment:
```suggestion
The SDK looks up the comparison attribute (the user attribute referenced by the condition) in the [User Object]. It compares the attribute value to the comparison value that is set on the Dashboard. The comparison is done according to the selected comparator, resulting in a `true` or `false` value. This will be the result of the condition.
```